client/scripts/DrawTogether.js

```diff
@@ -1295,6 +1326,262 @@ DrawTogether.prototype.handlePaintSelection = function handlePaintSelection (eve
+	if(newX !== null)
+		element.dataset.x = newX;
+	if(newY !== null)
+		element.dataset.y = newY;
+	if(newName !== null)
+		element.dataset.name = newName;
+	if(newOwner !== null)
+		element.dataset.owner = newOwner;
+	
+	var coordinateButton = element.getElementsByClassName("fav-coor-button")[0];
+	var inputRename = element.getElementsByClassName("fav-rename-input")[0];
+	if ( (newName || element.dataset.name) === ""){
+		inputRename.value = "";
+		coordinateButton.innerHTML = (newX || element.dataset.x) + "," + (newY || element.dataset.y);
+		coordinateButton.title = "";
+	}
+	else {
+		inputRename.value = (newName || element.dataset.name);
+		coordinateButton.innerHTML = (newName || element.dataset.name);
+		coordinateButton.title = (newX || element.dataset.x) + "," + (newY || element.dataset.y);
+	}
+	
+	
+};
+
+DrawTogether.prototype.insertOneFavorite = function insertOneFavorite(x, y, name, owner) {
+	var favoriteContainer = this.favoritesContainer.appendChild(document.createElement("div"));
+	favoriteContainer.className = "favorite-container";
+	favoriteContainer.dataset.x = x;
+	favoriteContainer.dataset.y = y;
+	favoriteContainer.dataset.name = name;
+	favoriteContainer.dataset.owner = owner;
+	
+	var favoriteMinusButton = favoriteContainer.appendChild(document.createElement("div"));
+	favoriteMinusButton.className = "fav-button fav-minus-button";
+	favoriteMinusButton.innerHTML = "-";
+	favoriteMinusButton.addEventListener("click", function (e) {
+		if(e.srcElement.classList.contains("fav-button-confirmation")){
+			var coord = e.srcElement.parentElement.getElementsByClassName("fav-coor-button")[0];
+			coord.dataset.tempConf = coord.innerHTML;
+			e.srcElement.innerHTML = "-";
+			e.srcElement.classList.remove("fav-button-confirmation");
+			
+			var greatgrandfather = e.srcElement.parentElement;
+			
+			var x = greatgrandfather.dataset.x;
+			var y = greatgrandfather.dataset.y;
+			var name = greatgrandfather.dataset.name;
+			this.removeFavorite(x, y, name, greatgrandfather);			
+		}
+		else {
+			var coord = e.srcElement.parentElement.getElementsByClassName("fav-coor-button")[0];
+			coord.dataset.tempConf = coord.innerHTML;
+			coord.innerHTML = "Delete?";
+			e.srcElement.classList.add("fav-button-confirmation");
+			e.srcElement.innerHTML = "?";
+			setTimeout(function() {
+				var coord = this.parentElement.getElementsByClassName("fav-coor-button")[0];
+				coord.innerHTML = coord.dataset.tempConf;
+				e.srcElement.innerHTML = "-";
+				e.srcElement.classList.remove("fav-button-confirmation");
+			}.bind(e.srcElement), 4000);
+		}
+	}.bind(this));
+	
+	var favoritePlusButton = favoriteContainer.appendChild(document.createElement("div"));
+	favoritePlusButton.className = "fav-button fav-plus-button";
+	favoritePlusButton.innerHTML = "+";
+	
+	favoritePlusButton.addEventListener("click", function (e) {
```

Comment:
Again moving this function to its own will make the insert function smaller and more readable.
Best to avoid having functions that have a lot of lines. Shoot for functions with 7-15 lines.